sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -220,41 +244,46 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
-    programOffering.educationProgram = { id: programId } as EducationProgram;
-    programOffering.institutionLocation = {
-      id: locationId,
-    } as InstitutionLocation;
-    programOffering.offeringIntensity =
-      educationProgramOffering.offeringIntensity;
-    programOffering.yearOfStudy = educationProgramOffering.yearOfStudy;
-    programOffering.showYearOfStudy = educationProgramOffering.showYearOfStudy;
-    programOffering.hasOfferingWILComponent =
-      educationProgramOffering.hasOfferingWILComponent;
-    programOffering.offeringWILType = educationProgramOffering.offeringWILType;
-    programOffering.studyBreaks = educationProgramOffering.breaksAndWeeks;
-    programOffering.offeringDeclaration =
-      educationProgramOffering.offeringDeclaration;
-    programOffering.offeringType = educationProgramOffering.offeringType;
-    programOffering.courseLoad = educationProgramOffering.courseLoad;
-    programOffering.offeringStatus = educationProgramOffering.offeringStatus;
+    if (!hasExistingApplication) {
+      programOffering.studyStartDate = educationProgramOffering.studyStartDate;
+      programOffering.studyEndDate = educationProgramOffering.studyEndDate;
+      programOffering.actualTuitionCosts =
+        educationProgramOffering.actualTuitionCosts;
+      programOffering.programRelatedCosts =
+        educationProgramOffering.programRelatedCosts;
+      programOffering.mandatoryFees = educationProgramOffering.mandatoryFees;
+      programOffering.exceptionalExpenses =
+        educationProgramOffering.exceptionalExpenses;
+      programOffering.offeringDelivered =
+        educationProgramOffering.offeringDelivered;
+      programOffering.lacksStudyBreaks =
+        educationProgramOffering.lacksStudyBreaks;
+      programOffering.offeringType =
+        educationProgramOffering.offeringType ?? OfferingTypes.Public;
+      programOffering.educationProgram = { id: programId } as EducationProgram;
+      programOffering.institutionLocation = {
+        id: locationId,
+      } as InstitutionLocation;
+      programOffering.offeringIntensity =
+        educationProgramOffering.offeringIntensity;
+      programOffering.yearOfStudy = educationProgramOffering.yearOfStudy;
+      programOffering.showYearOfStudy =
+        educationProgramOffering.showYearOfStudy;
+      programOffering.hasOfferingWILComponent =
+        educationProgramOffering.hasOfferingWILComponent;
+      programOffering.offeringWILType =
+        educationProgramOffering.offeringWILType;
+      programOffering.studyBreaks = educationProgramOffering.breaksAndWeeks;
+      programOffering.offeringDeclaration =
+        educationProgramOffering.offeringDeclaration;
+      programOffering.offeringType = educationProgramOffering.offeringType;
+      programOffering.courseLoad = educationProgramOffering.courseLoad;
+      programOffering.offeringStatus = educationProgramOffering.offeringStatus;
```

Comment:
not a blocker, since you are working the offering save method. can we add the logic to save the  audit fields

Reply:
Added logic to include audit user on create and update service.